app/storage/register-status-repository.js

```diff
@@ -0,0 +1,72 @@
+const { odata } = require('@azure/data-tables')
+const { tableClient } = require('./get-table-client')
+
+const maxDate = new Date(8640000000000000)
+
+const invertTimestamp = (timestamp) => {
+  const inverted = `${maxDate - timestamp}`
+
+  return inverted.padStart(19, '0')
+}
+
+const createEntity = (filename, email, status) => {
+  const now = new Date()
+
+  return {
+    PartitionKey: filename,
+    RowKey: invertTimestamp(now),
+    email,
+    status,
+    createdOn: now.toISOString(),
+    updatedAt: now.toISOString()
+  }
+}
+
+const addToTable = async (filename, email, status) => {
+  await tableClient.createTable()
+
+  const entity = createEntity(filename, email, status)
+
+  await tableClient.createEntity(entity)
+}
+
+const queryImportsByPartition = async (filename, additionalOptions) => {
+  await tableClient.createTable()
+
+  const results = tableClient.listEntities({
+    queryOptions: {
+      ...additionalOptions,
+      filter: odata`PartitionKey eq ${filename}`
+    }
+  })
+
+  return results
+}
+
+const setUploaded = async (filename, email) => await addToTable(filename, email, 'uploaded')
+
+const getLatestUpdate = async (filename) => {
+  const results = await queryImportsByPartition(filename, { top: 1 })
```

Comment:
Will this not return the first row, not necessarily the latest update?

Reply:
See above comment. Table storage doesn't provide a mechanism to orderBy so without the invertedTimestamp as row key we would have to get all entities in the partition and sort ourselves.

Using the log tail pattern, the `top: 1` will always return the smallest row key.